pkg/skaffold/initializer/init.go

```diff
@@ -35,35 +35,45 @@ import (
 
 // DoInit executes the `skaffold init` flow.
 func DoInit(ctx context.Context, out io.Writer, c config.Config) error {
+	var err error
 	rootDir := "."
 
 	if c.ComposeFile != "" {
-		if err := runKompose(ctx, c.ComposeFile); err != nil {
+		if err = runKompose(ctx, c.ComposeFile); err != nil {
 			return err
 		}
 	}
 
 	a := analyze.NewAnalyzer(c)
 
-	if err := a.Analyze(rootDir); err != nil {
+	if err = a.Analyze(rootDir); err != nil {
 		return err
 	}
 
 	buildInitializer := build.NewInitializer(a.Builders(), c)
+	deployInitializer := deploy.NewInitializer(a.Manifests(), c)
 
-	deployInitializer, err := deploy.NewInitializer(a.Manifests(), c)
-	if err != nil {
-		return err
-	}
-
-	if err := buildInitializer.ProcessImages(deployInitializer.GetImages()); err != nil {
+	if err = buildInitializer.ProcessImages(deployInitializer.GetImages()); err != nil {
 		return err
 	}
 
 	if c.Analyze {
 		return buildInitializer.PrintAnalysis(out)
 	}
 
+	var generatedManifests map[string][]byte
+	if c.EnableManifestGeneration {
+		generatedManifests, err = deployInitializer.GenerateManifests(buildInitializer.GeneratedPairs())
+		if err != nil {
+			return err
+		}
+		buildInitializer.Resolve()
```

Comment:
I am not a big fan of this pattern where the caller has to remember to call the function `Resolve`.
here IIUC, we generate manifest for generated pairs from user input?

This line could be easily removed in future and will affect the whole flow. Can we avoid this?

Reply:
I moved the actual manifest generation handling to the build initializer. now, the deploy initializer is responsible for tracking the generated manifests and image names, and the init controller itself is only responsible for passing those around. I think it's better this way.